### PR TITLE
iOS CI improvements

### LINKED ISSUE #607: Run iOS tests in CI
We started testing the iOS code in #606, but the tests aren't run as part of the CI process.

I don't know if its easier to do this using circleci or taskcluster. Probably circle, but doing that is (probably) blocked on us being able to use Macs in CI (no issue, but will probably be fixed as part of https://github.com/mozilla/application-services/issues/459).

### PR DESCRIPTION
**Pull request by eoger**
Fixes #607 #842

The last commit (`Use {{ epoch }} in the rust target cache key`) makes sure our Rust `target/` cache doesn't grow too big (I've noticed it's 5Gb now and takes a while to restore) by always uploading a new cache entry and letting CircleCI pick the most recent one on further runs.

### CODE REVIEW

File: megazords/ios/MozillaAppServicesTests/PlacesTests.swift [hunk 1]
```diff
@@ -1,8 +1,8 @@
+@testable import MozillaAppServices
```
Comment (thomcc) [inline megazords/ios/MozillaAppServicesTests/PlacesTests.swift]: This has to be above the license header? gross.

File: .circleci/config.yml [hunk 2]
```diff
@@ -197,16 +212,34 @@ jobs:
...
       - run:
-          name: Build Carthage archive
+          name: Setup build environment
           command: |
             rustup target add aarch64-apple-ios x86_64-apple-ios
             brew update
-            brew outdated carthage || brew upgrade carthage
             brew install swift-protobuf
+            # See https://circleci.com/docs/2.0/testing-ios/#pre-starting-the-simulator
+            xcrun instruments -w "iPhone 7 (12.1) [" || true
```
Comment (garvankeeley) [inline .circleci/config.yml]: TIL: this command!

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,6 +1,13 @@
 version: 2.1
 
 commands:
+  install-xcpretty:
+    steps:
+      - run:
+          name: Install xcpretty
+          command: |
+            gem install --user-install xcpretty
+            echo 'export PATH="$HOME/.gem/ruby/2.3.0/bin:$PATH"' >> $BASH_ENV
```
Comment (garvankeeley) [inline .circleci/config.yml]: Never seen this way of adding a path to the env, does this work?

File: .circleci/config.yml [hunk 4]
```diff
@@ -136,6 +143,14 @@ commands:
...
 
 jobs:
+  Check Swift formatting:
+    macos:
+      xcode: "10.1.0"
+    steps:
+      - checkout
+      - run: brew install swiftlint swiftformat
+      - run: swiftlint --strict
+      - run: swiftformat --lint --swiftversion 4 --verbose megazords
```
Comment (garvankeeley) [inline .circleci/config.yml]: I am not sure how formatting on the C.I. does anything, I am probably misunderstanding this step.

Comment (garvankeeley) [reply .circleci/config.yml]: oh nvm, --lint checks to see if swiftformat was run on the code and reports discrepancies.

File: .circleci/config.yml (hunk 3 above)
Comment (eoger) [reply .circleci/config.yml]: Yup that's the way CircleCI does env variables :/

**Pull request merged**